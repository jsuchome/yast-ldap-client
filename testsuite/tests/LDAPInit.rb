# encoding: utf-8

#  LDAPInit.ycp
#  Test of Ldap:LDAPInit, LDAPError, GetFirstServer, GetFirstPort functions
#  Author:	Jiri Suchomel <jsuchome@suse.cz>
#  $Id$
module Yast
  class LDAPInitClient < Client
    def main
      Yast.include self, "testsuite.rb"
      # testedfiles: Ldap.ycp

      @READ = {
        "target" => { "size" => -1 },
        "ldap"   => {
          "error"   => { "msg" => "Initialization failed", "code" => 11 },
          "product" => { "features" => { "EVMS_CONFIG" => "nazdar" } }
        }
      }
      @EX = { "ldap" => true }
      @EX_F = { "ldap" => false }

      TESTSUITE_INIT([@READ, {}, {}], nil)

      Yast.import "Ldap"

      DUMP("==== init (one server, no port set) ==============")

      Ldap.server = "localhost"

      DUMP(Builtins.sformat("==== value of server: \"%1\"", Ldap.server))

      TEST(lambda { Ldap.LDAPInit }, [{}, {}, @EX], 0)

      DUMP("==== init (one server, nonsence port set) ========")

      Ldap.server = "localhost:sdgfd\#$"

      DUMP(Builtins.sformat("==== value of server: \"%1\"", Ldap.server))

      TEST(lambda { Ldap.LDAPInit }, [{}, {}, @EX], 0)

      DUMP("==== init (more servers set, TLS used) ===========")

      Ldap.server = "chimera.suse.cz:333 localhost"
      Ldap.ldap_tls = true
      Ldap.tls_cacertdir = "/etc/ssl/certs"

      DUMP(Builtins.sformat("==== value of server: \"%1\"", Ldap.server))

      TEST(lambda { Ldap.LDAPInit }, [{}, {}, @EX], 0)

      DUMP("==== init failed =================================")

      TEST(lambda { Ldap.LDAPInit }, [@READ, {}, @EX_F], 0)

      nil
    end
  end
end

Yast::LDAPInitClient.new.main
