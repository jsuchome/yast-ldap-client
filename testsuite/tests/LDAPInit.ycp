/**
 *  LDAPInit.ycp
 *  Test of Ldap:LDAPInit, LDAPError, GetFirstServer, GetFirstPort functions
 *  Jiri Suchomel <jsuchome@suse.cz>
 *  $Id$
 */

{

    include "testsuite.ycp";
    // testedfiles: Ldap.ycp

    map READ =	$[ "ldap": $[
	"error": $[
	    "msg": "Initialization failed",
	    "code": 11
	],
	"product" : $[ "features" :
	    $[ "EVMS_CONFIG" : "nazdar" ],
	],
    ]];
    map EX =	$[ "ldap": true  ];
    map EX_F =	$[ "ldap": false ];

    TESTSUITE_INIT([READ,$[],$[]], nil);

    import "Ldap";

    DUMP ("==== init (one server, no port set) ==============");

    Ldap::server = "localhost";

    DUMP (sformat ("==== value of server: \"%1\"", Ldap::server));

    TEST(``(Ldap::LDAPInit ()), [ $[], $[], EX ], 0);

    DUMP ("==== init (one server, nonsence port set) ========");

    Ldap::server = "localhost:sdgfd#$";

    DUMP (sformat ("==== value of server: \"%1\"", Ldap::server));

    TEST(``(Ldap::LDAPInit ()), [ $[], $[], EX ], 0);

    DUMP ("==== init (more servers set, TLS used) ===========");

    Ldap::server = "chimera.suse.cz:333 localhost";
    Ldap::ldap_v2 = true;
    Ldap::ldap_tls = true;

    DUMP (sformat ("==== value of server: \"%1\"", Ldap::server));

    TEST(``(Ldap::LDAPInit ()), [ $[], $[], EX ], 0);

    DUMP ("==== init failed =================================");

    TEST(``(Ldap::LDAPInit ()), [ READ, $[], EX_F ], 0);

}
