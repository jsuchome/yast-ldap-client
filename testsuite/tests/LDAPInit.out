Dump	==== init (one server, no port set) ==============
Dump	==== value of server: "localhost"
Execute	.ldap $["hostname":"localhost", "port":389, "version":3] true
Return	
Dump	==== init (one server, nonsence port set) ========
Dump	==== value of server: "localhost:sdgfd#$"
Execute	.ldap $["hostname":"localhost", "port":0, "version":3] true
Return	
Dump	==== init (more servers set) =====================
Dump	==== value of server: "chimera.suse.cz:333 localhost"
Execute	.ldap $["hostname":"chimera.suse.cz", "port":333, "version":2] true
Return	
Dump	==== init failed =================================
Execute	.ldap $["hostname":"chimera.suse.cz", "port":333, "version":2] false
Read	.ldap.error $["code":11, "msg":"Initialization failed"]
Return	Initialization failed
